backend/main/server/app.go

```diff
@@ -756,36 +767,38 @@ func (a *App) createProposal(w http.ResponseWriter, r *http.Request) {
 		respondWithError(w, http.StatusForbidden, errMsg)
 		return
 	}
+
 	if err := a.validateSignature(p.Creator_addr, p.Timestamp, p.Composite_signatures); err != nil {
 		log.Error().Err(err)
 		respondWithError(w, http.StatusForbidden, err.Error())
 		return
 	}
 
-	// get latest snapshotted blockheight
-	snapshot, err := a.SnapshotClient.GetLatestSnapshot()
-	if err != nil {
-		log.Error().Err(err).Msg("error fetching latest snapshot")
-		respondWithError(w, http.StatusInternalServerError, err.Error())
-		return
-	}
-	p.Block_height = snapshot.Block_height
-
 	var community models.Community
 	community.ID = communityId
+
 	if err := community.GetCommunity(a.DB); err != nil {
 		log.Error().Err(err).Msg("error fetching community")
 		respondWithError(w, http.StatusInternalServerError, err.Error())
 		return
 	}
-
 	strategy, err := models.MatchStrategyByProposal(*community.Strategies, *p.Strategy)
 	if err != nil {
 		log.Error().Err(err).Msg("Community does not have this strategy availabe")
 		respondWithError(w, http.StatusInternalServerError, err.Error())
 		return
+
+	}
+	snapshot, err := a.SnapshotClient.GetLatestSnapshot(strategy.Contract)
```

Comment:
Replace this block w./ TakeSnapshot

Reply:
pesky merge conflict. Reverted back to its superior form ...